sdk/scheduler/src/main/java/com/mesosphere/sdk/config/validate/VolumePathCannotBeEmpty.java

```diff
@@ -0,0 +1,37 @@
+package com.mesosphere.sdk.config.validate;
+
+import com.mesosphere.sdk.specification.PodSpec;
+import com.mesosphere.sdk.specification.ServiceSpec;
+import com.mesosphere.sdk.specification.TaskSpec;
+import com.mesosphere.sdk.specification.VolumeSpec;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.*;
+
+/**
+ * Validates that volume path is not empty if new ServiceSpec has a volume
+ */
+public class VolumePathCannotBeEmpty implements ConfigValidator<ServiceSpec> {
+    @Override
+    public Collection<ConfigValidationError> validate(ServiceSpec nullableOldConfig, ServiceSpec newConfig) {
+        List<ConfigValidationError> errors = new ArrayList<>();
+
+        for (PodSpec podSpec : newConfig.getPods()) {
+            for (TaskSpec taskSpec : podSpec.getTasks()) {
+                Collection<VolumeSpec> volumeSpecs = taskSpec.getResourceSet().getVolumes();
+                if (volumeSpecs.isEmpty()) {
```

Comment:
get ResourceSet returns a Collection. If no VolumeSpec, it should be an empty list.